src/AzureAuth/AuthModeExtensions.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth
+{
+    using Microsoft.Authentication.MSALWrapper;
+    using Microsoft.Office.Lasso.Interfaces;
+
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Extensions to <see cref="AuthMode"/>s.
+    /// </summary>
+    public static class AuthModeExtensions
+    {
+        /// <summary>
+        /// Filters a list of <see cref="AuthMode"/> for interaction returning a single aggregate <see cref="AuthMode"/>.
+        /// </summary>
+        /// <param name="authModes"><see cref="AuthMode"/>s to aggregate and filter.</param>
+        /// <param name="env">An <see cref="IEnv"/> to use.</param>
+        /// <returns>A single aggregate <see cref="AuthMode"/> filtered for interactivity.</returns>
+        public static AuthMode FilterInteraction(this IEnumerable<AuthMode> authModes, IEnv env)

```

Comment:
I think there's room to refactor/rename this method, but let's stick to mechanical transformations for now and keep this as `CombinedAuthMode`.